packages/engine/lib/error/src/lib.rs

```diff
@@ -228,23 +224,69 @@ pub use self::{
     result::{IntoReport, Result, ResultExt},
 };
 
-/// Trait alias for an error context.
+/// Defines the current context of a [`Report`].
 ///
-/// Note: This is currently defined as trait but will be changed to be a trait alias as soon as
-///   it's stabilized
-// TODO: change to `pub trait Context = ...`
-//   Tracking issue: https://github.com/rust-lang/rust/issues/41517
-pub trait Context: Provider + fmt::Display + fmt::Debug + Send + Sync + 'static {}
-impl<C: Provider + fmt::Display + fmt::Debug + Send + Sync + 'static> Context for C {}
-
-/// Trait alias for a contextual message.
+/// Every [`Error`] also can act as a `Context`. If [`Error`] is not implemented or in a non-std
+/// environment, this trait can be implemented manually. In most cases, implementing this is not
+/// needed.
+///
+/// [`Error`]: std::error::Error
+///
+/// ## Example
+///
+/// Used for creating a [`Report`] or for switching the [`Report`]s context:
+///
+/// ```rust
+/// # #![cfg_attr(any(not(feature = "std"), miri), allow(unused_imports))]
+/// use std::{fmt, fs, io};
+///
+/// use error::{Context, IntoReport, Result, ResultExt};
+///
+/// # type Config = ();
+/// #[derive(Debug)]
+/// pub enum ConfigError {
+///     ParseError,
+/// }
+///
+/// impl fmt::Display for ConfigError {
+///     # #[allow(unused_variables)]
+///     fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+///         # const _: &str = stringify! {
+///         ...
+///         # }; Ok(())
+///     }
+/// }
+///
+/// // `Error` is not implemented for `ConfigError` for any reason, so implement `Context` manually.
+/// impl Context for ConfigError {}
+///
+/// # #[cfg(any(not(feature = "std"), miri))]
+/// # pub fn read_file(_: &str) -> Result<String, ConfigError> { error::bail!(ConfigError::ParseError) }
+/// # #[cfg(all(feature = "std", not(miri)))]
+/// pub fn read_file(path: &str) -> Result<String, io::Error> {
+///     // Creates a `Report` from `io::Error`, the current context is `io::Error`
+///     fs::read_to_string(path).report()
+/// }
+///
+/// pub fn parse_config(path: &str) -> Result<Config, ConfigError> {
+///     // The return type of `parse_config` requires another context. By calling `change_context`
+///     // the context may be changed.
+///     read_file(path).change_context(ConfigError::ParseError)?;
 ///
-/// Note: This is currently defined as trait but will be changed to be a trait alias as soon as
-///   it's stabilized
-// TODO: change to `pub trait Context = ...`
-//   Tracking issue: https://github.com/rust-lang/rust/issues/41517
-pub trait Message: fmt::Display + fmt::Debug + Send + Sync + 'static {}
```

Comment:
Previously, we had two traits "aliases", now we have `Message`, which is an alias, and `Context`, which is not. As users eventually have to implement `Context`, having another trait, which doesn't need to be implemented, may be confusing. Thus, deciding to remove the alias.

Reply:
I was going to ask, I think the reasoning makes sense although the trait expressions over the place are pretty unwieldy. We can always add it back if need be